### ISSUE TITLE
`numpy.ma.masked_where()` return object not recognized as `numpy.ma.MaskedArray`

### ISSUE THREAD

**Issue by yumasheta** [PyCQA/pylint#3342]
## Steps to reproduce
Consider this short python script:
```python
"""Quick test to show pylint error."""

import numpy as np

DATA = np.array([1, 2, 3])
DATA = np.ma.masked_where([1, 0, 0], DATA)
print(DATA.mask)
assert isinstance(DATA, np.ma.MaskedArray)

DATA = np.ma.array([1, 2, 3], mask=[1, 0, 0])
print(DATA.mask)
assert isinstance(DATA, np.ma.MaskedArray)
```
With output:
```
[ True False False]
[ True False False]
```
Running pylint on that code returns:
```
7:6: E1101: Instance of 'ndarray' has no 'mask' member (no-member)
```
## Current behavior
The object returned by `numpy.ma.masked_where()` is an instance of `numpy.ma.MaskedArray`, just as the object returned by `numpy.ma.array()` is. However, pylint does not recognize the return value of `numpy.ma.masked_where()` as such.

### Expected behavior

Pylint should recognize that `numpy.ma.masked_where()` returns a `numpy.ma.MaskedArray`.

### pylint --version output
```
pylint 2.4.4
astroid 2.3.3
Python 3.8.1 (default, Dec 21 2019, 20:57:38) 
[GCC 9.2.0]
```

**Comment by hippo91** [comment]
@yumasheta thanks for this bug report. I'll investigate ASAP.

**Comment by hippo91** [comment]
@yumasheta the problem comes from the fact that there is not a brain for `numpy.ma` module in astroid yet. I put it on my todo list but i'm really busy right now so it will take a long time before seeing it in `astroid`.
Feel free to add it through a PR. I'll be happy to help you if needed.

**Comment by hippo91** [comment]
@yumasheta this bug should be fixed once PyCQA/astroid#1148 will be merged and by authorizing `pylint` to dynamically load `numpy` with `--extension-pkg-allow-list=numpy` option.
If however you don't want to authorize this loading, then an improvement of `astroid`'s `numpy` brain is required but may take a very long time to be set up.

**Comment by hippo91** [comment]
I'm reopening the issue because what was made in PyCQA/astroid#1148 was not wise.

**Comment by hippo91** [comment]
The problem will be fixed definitively through an update of astroid numpy brain modules.
